src/pylhe/__init__.py

```diff
@@ -191,20 +199,22 @@ def readLHEWithAttributes(thefile):
                 for p in particles:
                     if not p.strip().startswith("#"):
                         eventdict["particles"] += [LHEParticle.fromstring(p)]
+                    elif parse_optional:
+                        if "optional" not in list(eventdict.keys()):
+                            eventdict["optional"] = []
```

Comment:
maybe this could be added as a default empty list in the eventdict, regardless of the `parse_optional`?

Reply:
See [1c8e8df](https://github.com/scikit-hep/pylhe/pull/28/commits/1c8e8df50922317df9fc88369c912df5d8fade28)